File: linked_list_ToBinary/binary.go
package linkedlisttobinary

import "strconv"

type ListNode struct {
	Val  int
	Next *ListNode
}

func getDecimalValue(head *ListNode) int {
	binaryStr := ""

	for head != nil {
		binaryStr += strconv.Itoa(head.Val)
		head = head.Next
	}

	intVal, err := strconv.ParseInt(binaryStr, 2, 64)
	if err != nil {
		return 0
	}

	return int(intVal)
}

File: linked_list_ToBinary/binary.js
function ListNode(val, next) {
	this.val = val === undefined ? 0 : val;
	this.next = next === undefined ? null : next;
}
/**
 * @param {ListNode} head
 * @returns {number}
 */
var getDecimalValue = function (head) {
	let binaryStr = "";

	while (head !== null) {
		binaryStr += head.val;
		head = head.next;
	}

	return parseInt(binaryStr, 2);
};
